misk/src/main/kotlin/misk/client/HttpClientsConfig.kt

```diff
@@ -1,115 +1,75 @@
 package misk.client
 
+import com.fasterxml.jackson.annotation.JsonAlias
 import com.fasterxml.jackson.databind.annotation.JsonDeserialize
-import com.fasterxml.jackson.databind.type.TypeFactory
 import misk.config.Config
+import misk.logging.getLogger
 import misk.security.ssl.CertStoreConfig
 import misk.security.ssl.TrustStoreConfig
 import java.io.File
+import java.net.URL
 import java.time.Duration
 
-data class BackwardsCompatibleEndpointConfig(
-    //Common fields
-  val url: String? = null,
-  val envoy: HttpClientEnvoyConfig? = null,
-
-    //Legacy fields
-  val connectTimeout: Duration? = null,
-  val writeTimeout: Duration? = null,
-  val readTimeout: Duration? = null,
-  val pingInterval: Duration? = null,
-  val callTimeout: Duration? = null,
-  val maxRequests: Int = 128,
-  val maxRequestsPerHost: Int = 32,
-  val maxIdleConnections: Int = 100,
-  val keepAliveDuration: Duration = Duration.ofMinutes(5),
-  val ssl: HttpClientSSLConfig? = null,
-
-    //New fields
-  val clientConfig: HttpClientConfig? = null
-)
-
-class BackwardsCompatibleEndpointConfigConverter : com.fasterxml.jackson.databind.util.Converter<BackwardsCompatibleEndpointConfig, HttpClientEndpointConfig> {
-  override fun getInputType(typeFactory: TypeFactory) =
-      typeFactory.constructType(BackwardsCompatibleEndpointConfig::class.java)
-
-  override fun getOutputType(typeFactory: TypeFactory) =
-      typeFactory.constructType(HttpClientEndpointConfig::class.java)
-
-  override fun convert(value: BackwardsCompatibleEndpointConfig) =
-      //http://%s.%s.gns.square
-      if (value.clientConfig == null)
-        HttpClientEndpointConfig(
-            url = value.url,
-            envoy = value.envoy,
-            clientConfig = HttpClientConfig(
-                value.connectTimeout,
-                value.writeTimeout,
-                value.readTimeout,
-                value.pingInterval,
-                value.callTimeout,
-                value.maxRequests,
-                value.maxRequestsPerHost,
-                value.maxIdleConnections,
-                value.keepAliveDuration,
-                value.ssl
-            )
-        )
-      else
-        HttpClientEndpointConfig(
-            url = value.url,
-            envoy = value.envoy,
-            clientConfig = value.clientConfig
-        )
-}
-
+@JsonDeserialize(converter = BackwardsCompatibleClientsConfigConverter::class)
```

Comment:
The backwards compatibility converter moved from per-endpoint, to `HttpClientsConfig` given I want to also deprecate the `default*` attributes.